### PR TITLE
Updating ROS.19 validation to make it more robust

### PR DESCRIPTION
**Pull request by allenbross-wf**
#### Reason for change

ROS.19 was having an issue for facts without units.

#### Description of change

Checks for existence of units before inspecting them.

#### Steps to Test

Validate a ROS document that has monetary facts without units.

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary3-wk** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

**Review by derekgengenbacher-wf — DISMISSED** [review]

File: arelle/plugin/validate/ROS/rules/ros.py [hunk 1]
```diff
@@ -354,9 +354,9 @@ def rule_ros19(
     message: str | None = None
     modelObjects = set()
     monetaryFacts = val.modelXbrl.factsByDatatype(False, qnXbrliMonetaryItemType)
-    monetaryUnits = set(fact.unit.value for fact in monetaryFacts)
+    monetaryUnits = set(fact.unit.value if fact.unit and fact.unit.value else None for fact in monetaryFacts)
```
Comment (brettkail-wk) [inline arelle/plugin/validate/ROS/rules/ros.py:357]: Why `and fact.unit.value`?

It seems like this should exclude the unit altogether rather than produce a `None` (the user will already separately get an error for a missing unit on a monetary fact):
```
set(fact.unit.value for fact in monetaryFacts if fact.unit)
```

File: arelle/plugin/validate/ROS/rules/ros.py [hunk 2]
```diff
@@ -354,9 +354,9 @@ def rule_ros19(
...
     pricipalCurrencyFacts = val.modelXbrl.factsByLocalName.get(PRINCIPAL_CURRENCY, set())
-    principalCurrencyValues = set(fact.text for fact in pricipalCurrencyFacts)
+    principalCurrencyValues = set(fact.text if fact.text else None for fact in pricipalCurrencyFacts)
```
Comment (brettkail-wk) [inline arelle/plugin/validate/ROS/rules/ros.py:359]: Why this change?  If xsi:nil or empty string, it seems like those _should_ be added to the set to force a mismatch.

**Review by derekgengenbacher-wf — APPROVED** [review]

**Review by aaroncameron-wk — APPROVED** [review]

**Comment by derekgengenbacher-wf** [comment]
QA +10 Testing locally succeeded.
